orchagent/macsecorch.cpp

```diff
@@ -0,0 +1,2283 @@
+#include "macsecorch.h"
+
+#include <macaddress.h>
+#include <redisapi.h>
+#include <redisclient.h>
+#include <sai_serialize.h>
+
+#include <vector>
+#include <sstream>
+#include <algorithm>
+#include <functional>
+#include <stack>
+#include <memory>
+
+/* Global Variables*/
+
+#define AVAILABLE_ACL_PRIORITIES_LIMITATION             (32)
+#define EAPOL_ETHER_TYPE                                (0x888e)
+#define MACSEC_STAT_FLEX_COUNTER_POLLING_INTERVAL_MS    (1000)
+#define COUNTERS_MACSEC_ATTR_GROUP                      "COUNTERS_MACSEC_ATTR"
+
+extern sai_object_id_t   gSwitchId;
+extern sai_macsec_api_t *sai_macsec_api;
+extern sai_acl_api_t *sai_acl_api;
+extern sai_port_api_t *sai_port_api;
+extern sai_switch_api_t *sai_switch_api;
+
+static const std::vector<std::string> macsec_egress_sa_stats =
```

Comment:
When you add support for MACSEC_SA_STATS, this will need to be renamed since it is not a vector of stats but attrs.